File: 09-HASHING/Go/contains-duplicate.go
package main

// https://leetcode.com/problems/contains-duplicate/

func containsDuplicate(nums []int) bool {
	h := map[int]bool{}
	for _, num := range nums {
		if h[num] {
			return true
		}
		h[num] = true
	}
	return false
}

File: 09-HASHING/PHP/contains-duplicate.php
<?php

// https://leetcode.com/problems/contains-duplicate/

class Solution
{
    function containsDuplicate(array $nums): bool
    {
        $h = [];
        foreach ($nums as $num) {
            if (key_exists($num, $h)) {
                return false;
            }
            $h[$num] = 1;
        }

        return true;
    }
}